worker/internal/app/handler.go

```diff
@@ -39,16 +53,30 @@ func (h Handler) DecompressHandler() echo.HandlerFunc {
 
 func (h Handler) WebhookHandler() echo.HandlerFunc {
 	return func(c echo.Context) error {
-		var msg msgBody
 		var w webhook.Webhook
-		if err := c.Bind(&msg); err != nil {
-			if err := c.Bind(&w); err != nil {
+		if c.Request().Header.Get("X-Amz-Sns-Message-Type") == "Notification" {
+			var payload sns.Payload
+			if err := json.NewDecoder(c.Request().Body).Decode(&payload); err != nil {
+				return err
+			}
+			if err := json.Unmarshal([]byte(payload.Message), &w); err != nil {
+				return err
+			}
+			// Validates payload's signature
+			if err := payload.VerifyPayload(); err != nil {
+				return err
+			}
+		} else {
+			var msg msgBody
```

Comment:
If you intend to use PubSub, e.g. `else if isGCP() ...` is required. Also separate if else blocks into functions:

```go
func (h Handler) isSNS(r *http.Request) bool {

}

func (h Handler) isGCP(r *http.Request) bool {

}

func parseSNSMessage(body io.Reader) (webhook.Webhook, error) {

}

func parsePubSubMessage(body io.Reader) (webhook.Webhook, error) {

}
```

Reply:
Done!